syncstorage/storage/memcached.py

```diff
@@ -762,7 +767,7 @@ def _set_items(self, userid, items, modified, data, casid):
                 if "payload" not in bso:
                     bso["payload"] = ""
                 data["items"][bso["id"]] = bso
-                data["modified"] = modified
+            data["modified"] = modified
```

Comment:
This is the buggy line of code - we should update the "modified" timestamp whenever any items change, but this indentation meant it was only updated when you add new items.